src/pymor/parameters/base.py

```diff
@@ -122,43 +129,58 @@ def fail(msg):
             raise ValueError(f'{mu_str} is incompatible with Parameters {self} ({msg})')
 
         if not self:
-            mu is None or mu == {} or fail('must be None or empty dict')
+            mu is None or len(mu) == 0 or fail('must be None or empty dict')
             return Mu({})
 
-        elif isinstance(mu, Mu):
+        if isinstance(mu, Mu):
             mu.parameters == self or fail(self.why_incompatible(mu))
             set(mu) == set(self) or fail(f'additional parameters {set(mu) - set(self)}')
             return mu
 
-        elif isinstance(mu, Number):
-            1 == sum(v for v in self.values()) or fail('need more than one number')
-            return Mu({next(iter(self)): np.array([mu])})
+        # convert mu to dict
+        if isinstance(mu, (Number, str, Function)):
+            assert len(self) == 1 or fail('not enough values')
+            mu = {next(iter(self.keys())): mu}
 
         elif isinstance(mu, (tuple, list, np.ndarray)):
             if isinstance(mu, np.ndarray):
                 mu = mu.ravel()
-            all(isinstance(v, Number) for v in mu) or fail('not every element a number')
-            len(mu) == sum(v for v in self.values()) or fail('wrong size')
+            all(isinstance(v, (Number, str, Function)) for v in mu) or \
+                fail('not every element a number or function')
             parsed_mu = {}
             for k, v in self.items():
-                p, mu = mu[:v], mu[v:]
+                len(mu) > 0 or fail('not enough values')
+                if isinstance(mu[0], (str, Function)):
+                    p, mu = mu[0], mu[1:]
```

Comment:
How are vector-valued functions parsed?

Reply:
The rule is that every single parameter is either a numpy array of shape `(d,)` or a Function with `shape_range == (d,)`. If `parse` encounters a `Function`, it is assumed it is the `Function` for the entire parameter.

Reply:
Ok, I was actually wondering about strings, but didn't make it clear. How should vector-valued functions be given as strings? Can one string per component be used?

Reply:
A string is treated the same way as a `Function` except that the string is first converted into an `ExpressionFunction`. Thus, strings also have to be one string for all components. When #1277 is merged, it is possible from the expression to obtain the shape of the range. Then it would be possible to have one expression per component.

Reply:
Ok, sounds good.

Reply:
I have opened #1396 as a reminder.